src/main.ts

```diff
@@ -1,74 +1,41 @@
+import { DEFAULT_SETTINGS } from 'defaults'
+import { Box, CalendarData, CalendarSettings, Entry } from 'model'
 import { Plugin } from 'obsidian'
 import HeatmapCalendarSettingsTab from 'settings'
 
-interface CalendarData {
-    year: number
-    colors:
-        | {
-              [index: string | number]: string[]
-          }
-        | string
-    entries: Entry[]
-    showCurrentDayBorder: boolean
-    defaultEntryIntensity: number
-    intensityScaleStart: number
-    intensityScaleEnd: number
-}
-
-interface CalendarSettings extends CalendarData {
-    colors: {
-        [index: string | number]: string[]
+declare global {
+    interface Window {
+        renderHeatmapCalendar: (el: HTMLElement, calendarData: CalendarData) => void
     }
-    weekStartDay: number
 }
 
-interface Entry {
-    date: string
-    intensity?: number
-    color: string
-    content: string
-}
-const DEFAULT_SETTINGS: CalendarSettings = {
-    year: new Date().getFullYear(),
-    colors: {
-        default: ['#c6e48b', '#7bc96f', '#49af5d', '#2e8840', '#196127'],
-    },
-    entries: [{ date: '1900-01-01', color: '#7bc96f', intensity: 5, content: '' }],
-    showCurrentDayBorder: true,
-    defaultEntryIntensity: 4,
-    intensityScaleStart: 1,
-    intensityScaleEnd: 5,
-    weekStartDay: 1,
-}
 export default class HeatmapCalendar extends Plugin {
-    settings: CalendarSettings
```

Comment:
removing this since it is not needed, and this way we can enable strictPropertyInitialization in tsconfig